packages/core-ethereum/src/index.ts

```diff
@@ -368,13 +369,18 @@ export default class HoprCoreEthereum extends EventEmitter {
         )
 
         log(ticket.ticket.toString())
-        const result = await boundRedeemTicket(channel.source, ticket)
+
+        const result = await boundRedeemTicket(channel.source, channelId, ticket)
 
         if (result.status !== 'SUCCESS') {
           if (result.status === 'ERROR') {
             // We need to abort as tickets require ordered redemption.
             // delete operation before returning
             throw result.error
+          } else {
+            // May fail due to out-of-commits, preimage-is-empty, not-a-winning-ticket
```

Comment:
This is not in-line with the cases where something other than `ERROR` is returned it seems. Could you double check that this aligns?

Reply:
Hold on... `FAILURE` can only happen before an on-chain redemption; `SUCCESS` and `ERROR` are returned depending on the result of on-chain redemption.
Here the code only deletes acked tickets when `FAILURE` is returned. `SUCCESS` has already been handled in `redeemTicket()` as acked tickets are marked as `redeemed` then get deleted. When `ERROR` returns (where the on-chain tx failed), tickets are NOT deleted, because we may allow a second attempt of submission.
Should we treat `ERROR` differently?

Reply:
> This is not in-line with the cases where something other than `ERROR` is returned it seems. Could you double check that this aligns?

You're right but this is out of scope for now

>  Should we treat ERROR differently?

Yes, but out of scope for now